src/Symfony/Component/Mailer/TransportFactory.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer;
+
+use Symfony\Component\Mailer\Exception\UnsupportedHostException;
+use Symfony\Component\Mailer\Transport\Dsn;
+use Symfony\Component\Mailer\Transport\TransportFactoryInterface;
+use Symfony\Component\Mailer\Transport\TransportInterface;
+
+/**
+ * @author Fabien Potencier <fabien@symfony.com>
+ * @author Konstantin Myakshin <molodchick@gmail.com>
+ */
+class TransportFactory
```

Comment:
What about keeping the `Transport` class name (I don't like classes with a name that end with `Factory`)? And if `$factories` is empty, fall back to the built-in transports. That way, we keep the current simple way of creating a Transport (without using Symfony full-stack) and we keep BC as well.